src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+    data_array: DataArray,
+    expected_cell_size: int,
+    categorical: bool,
+) -> DataArray:
+    """Returns a `DataArray` with scaled/resampled data using OpenCV.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
+        categorical (bool): Determines which resampling/interpolation is used.
+            Set to False will use bilinear. Set to True will use nearest-neighbour.
+
+    Returns:
+        DataArray
+    """
+    if categorical:
+        interpolation = INTER_NEAREST
+    else:
+        interpolation = INTER_LINEAR
+    resampled = resize(
+        src=data_array.data,
+        dsize=_get_resample_shape(data_array, expected_cell_size)[::-1],
+        interpolation=interpolation,
+    )
+
+    return _create_data_array(
+        data_array=data_array,
+        data=resampled,
+        geotransform=(
+            f"{data_array.rio.transform().c} {expected_cell_size} 0 "
+            f"{data_array.rio.transform().f} 0 -{expected_cell_size}"
+        ),
+    )
+
+
+def _check_no_data(
+    data_array: DataArray,
+    nodata: Optional[float],
+) -> None:
+    """Checks and conditionally writes a `nodata` value to data_array.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        nodata (Optional[float]): Value that will be written to the
+            input `DataArray` if it is not `None`.
+
+    Raises:
+        Exception: If the `nodata` arg is `None` and the input `DataArray`
+            does not have a nodata value (that is not equal to None).
+    """
+    if nodata is not None:
+        data_array.rio.write_nodata(nodata, inplace=True)
+    elif data_array.rio.nodata is None:
+        raise Exception(
+            "Input dataset does not have a nodata value. One must be provided."
+        )
+
+
+def _get_origin_offset(
+    data_array: DataArray,
+    expected_x_min: int,
+    expected_y_max: int,
+) -> NDArray:
+    """Determines the offset of the input `DataArray` from the origin (ymax, xmin).
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_x_min (int): x minimum.
+        expected_y_max (int): y maximum.
+
+    Returns:
+        NDArray: 2d array of the offset, (height, width).
+    """
+    return (
+        (
+            array(
+                [
+                    expected_y_max - data_array.rio.transform().f,
+                    expected_x_min + data_array.rio.transform().c,
+                ]
+            )
+            / int(data_array.rio.transform().a)
+        )
+        .round()
+        .astype("int")
+    )
+
+
+def _to_grid_extent(
+    data_array: DataArray,
+    expected_x_min: int,
+    expected_x_max: int,
+    expected_y_max: int,
+    nodata: Optional[float],
+) -> DataArray:
+    """Returns a `DataArray` where the data of the input (`data_array`) has been \
+    inserted into a grid of shape (`expected_y_max`, `expected_x_max`).
```

Comment:
As above, the first line of a function docstring should should be a one line (<88 character) description of the function.

```suggestion
    """Inserts the input `data_array` into a `expected_y_max` * `expected_x_max` grid.
```